libs/s25main/BasePlayerInfo.cpp

```diff
@@ -4,24 +4,25 @@
 
 #include "BasePlayerInfo.h"
 #include "helpers/serializeEnums.h"
-#include "s25util/Serializer.h"
 #include "s25util/colors.h"
 
 BasePlayerInfo::BasePlayerInfo()
-    : ps(PlayerState::Free), nation(Nation::Romans), color(PLAYER_COLORS[0]), team(Team::None)
+    : ps(PlayerState::Free), portraitIndex(0), nation(Nation::Romans), color(PLAYER_COLORS[0]), team(Team::None)
 {}
 
-BasePlayerInfo::BasePlayerInfo(Serializer& ser, bool lightData)
+BasePlayerInfo::BasePlayerInfo(Serializer& ser, int serializedVersion, bool lightData)
     : ps(helpers::popEnum<PlayerState>(ser)), aiInfo(!lightData || ps == PlayerState::AI ? ser : AI::Info())
 {
     if(lightData && !isUsed())
     {
+        portraitIndex = 0;
         nation = Nation::Romans;
         team = Team::None;
         color = PLAYER_COLORS[0];
     } else
     {
         name = ser.PopLongString();
+        portraitIndex = (serializedVersion >= 1) ? ser.PopUnsignedInt() : 0;
```

Comment:
Please add verification here (see discussion: Every point where it is set should be verified.). Could be just a `std::min`

Reply:
Done! the BasePlayerInfo.cpp was complaining about unknown std::array, so I've refactored the array definition to make_array instead of just adding the header in PortraitConsts.